Non-Blocking Upload & Progress Bar

ProjectBarks opened the pull request:
# Major
- Uploading occurs on separate thread and cleans up upon failure
- Uploading will replace the current UI with a progress bar made from a fake operator & depressed button. 
- Added simple packager 
- Moved module to subdirectory

# Minor
- Renamed `upload_gltf.py` to `upload.py`
- Separated uploading into `UploadManager` and handling user upload into `UploadOperator`

ProjectBarks committed:
rename upload_gltf to upload

ProjectBarks committed:
fix rename

ProjectBarks committed:
remove threading

ProjectBarks committed:
Add progress bar and run upload on separate thread

ProjectBarks committed:
Merged beta and master

ProjectBarks committed:
restructure folders

ProjectBarks committed:
Packager + Assets

Added packager which will build a zip and check the latest version has been changed

default testing assets are now built into the repo for pre release testing

ProjectBarks committed:
filter from git ignored file

ProjectBarks commented:
@mramato

ProjectBarks committed:
Update packaging to wrap zip in folder

ProjectBarks committed:
Add Documentation - Release Instructions

Wrote release instructions and added folder for those instructions

ProjectBarks committed:
add download button

tfili commented:
I tested this and it works. The code looks ok (at least I think, I've never written a line of Python in my life). @mramato Do you want to have a look?

mramato commented:
Works great, a couple of comments.

What are the licenses for the test models you added and where did they come from?  This information should be included in a ThirdParty.md.  We should also have an entry in there for whatever top-level package we are including in the third_party folder.

mramato commented on documentation/release-instructions.md:
```diff
@@ -0,0 +1,30 @@
+Releases are available on the [downloads page](https://github.com/AnalyticalGraphicsInc/ion-blender-exporter/releases). There is no release manager; instead, our community shares the responsibility.  Any committer can create the release for a given month, and at any point, they can pass the responsibility to someone else, or someone else can ask for it.  This spreads knowledge, avoids stratificat ...
+## Packaging, Testing, Release
+### Package
+1. Send a courtesy message in the [slack](https://cesium.slack.com) to let people know you're about to start the release process.
```
Can probably remove all items about slack, since it's not a public channel.

ProjectBarks replied:
@mramato Slack is in the cesium release notes?

ProjectBarks replied:
> Works great, a couple of comments.
> 
> What are the licenses for the test models you added and where did they come from? This information should be included in a ThirdParty.md. We should also have an entry in there for whatever top-level package we are including in the third_party folder.

They have always been there. I just moved them.

mramato replied:
Okay, sorry I missed it the first time.  We still need to properly document them

ProjectBarks replied:
I was thinking and I will remove them from the repo but have it be a part of the developer setup process.

mramato replied:
Maybe I'm missing something, but I don't see any mention here: https://github.com/AnalyticalGraphicsInc/cesium/wiki/Release-Guide I don't think we reference it in any open source project (but I could be wrong)

ProjectBarks replied:
My bad mistook forum as slack `Send a courtesy message to the forum to let people know you're about to start the release process.`

mramato replied:
Yeah, these smaller projects can have a little lighter of a release process. No big deal.

mramato commented on documentation/release-instructions.md:
```diff
@@ -0,0 +1,30 @@
+         - For `io-cesium-ion-v0.0.zip` the tag would be `v0.0`
+    - In the release add a change-log detailing what has been added. This should be as simple as copying the change-log from the merged pull requests.
+1. Upload the packaged file (ie `io-cesium-ion-v0.0.zip`) that was created in the packaging step.
+1. Notify everyone on [slack](https://cesium.slack.com) that the release is being published.
```
After removing this item, add an item at the end about making sure the new release is included in the monthly CesiumJS release announcement/blog post.

ProjectBarks committed:
Remove third_party

the intent is to remove third_party so it is apart of the installation process rather than cloned with the repo

ProjectBarks committed:
Updated release information

Added development docs

Made installing vendors easy

ProjectBarks commented:
@mramato

ProjectBarks committed:
Fixed vendor packing

mramato commented:
Any ideas why `python3 utils.py vendor` is failing?

```
Checking for old third_party...
Setting up third_party
Installing vendor (Ignore non-exitting errors)...
Exception:
Traceback (most recent call last):
  File "/usr/lib/python3/dist-packages/pip/basecommand.py", line 215, in main
    status = self.run(options, args)
  File "/usr/lib/python3/dist-packages/pip/commands/install.py", line 360, in run
    prefix=options.prefix_path,
  File "/usr/lib/python3/dist-packages/pip/req/req_set.py", line 784, in install
    **kwargs
  File "/usr/lib/python3/dist-packages/pip/req/req_install.py", line 851, in install
    self.move_wheel_files(self.source_dir, root=root, prefix=prefix)
  File "/usr/lib/python3/dist-packages/pip/req/req_install.py", line 1064, in move_wheel_files
    isolated=self.isolated,
  File "/usr/lib/python3/dist-packages/pip/wheel.py", line 247, in move_wheel_files
    prefix=prefix,
  File "/usr/lib/python3/dist-packages/pip/locations.py", line 153, in distutils_scheme
    i.finalize_options()
  File "/usr/lib/python3.6/distutils/command/install.py", line 274, in finalize_options
    raise DistutilsOptionError("can't combine user with prefix, "
distutils.errors.DistutilsOptionError: can't combine user with prefix, exec_prefix/home, or install_(plat)base
Traceback (most recent call last):
  File "utils.py", line 148, in <module>
    install_third_party(module_dir)
  File "utils.py", line 130, in install_third_party
    cwd=vendor_dir)
  File "/usr/lib/python3.6/subprocess.py", line 336, in check_output
    **kwargs).stdout
  File "/usr/lib/python3.6/subprocess.py", line 418, in run
    output=stdout, stderr=stderr)
subprocess.CalledProcessError: Command '['/usr/bin/python3', '-m', 'pip', 'install', 'boto3', '-t', '/home/mamato/Git/AnalyticalGraphicsInc/ion-blender-exporter/io-cesium-ion/third_party']' returned non-zero exit status 2.
```

ProjectBarks committed:
fixed install

mramato commented:
Still having issues

> $ python3 utils.py vendor
Checking for old third_party...
Installing vendor (Ignore non-exitting errors)...
/usr/lib/python3/dist-packages/pip/commands/install.py:212: UserWarning: Disabling all use of wheels due to the use of --build-options / --global-options / --install-options.
  cmdoptions.check_install_build_global(options)
Cleaning up...
Traceback (most recent call last):
  File "utils.py", line 149, in <module>
    install_third_party(module_dir)
  File "utils.py", line 130, in install_third_party
    packages_dir = next(Path(tmp_vendor_dir).glob("**/site-packages"))
StopIteration

ProjectBarks committed:
Unable to automatically install vendor

ProjectBarks commented:
Created https://github.com/AnalyticalGraphicsInc/ion-blender-exporter/issues/6 to document bug but going for hardcoded vendor

mramato commented:
As a general rule, we don't merge experimental/TODO code into master, that's what source control is for.  Please remove the vendor stuff (you can always park the code on a branch for when we go back to it).

ProjectBarks committed:
remove vendor code

mramato commented:
Using https://github.com/AnalyticalGraphicsInc/ion-sketchup-exporter/pull/12/files as a template, please add ThirdParty information for the vendor packages.

Also, please at a README to the test-assets folder with license and url information for each of the models so that we know where they came from.

ProjectBarks committed:
update test asset legals

add third party docs

ProjectBarks committed:
fix typo

ProjectBarks committed:
test asset cleanup

mramato commented:
Thanks.  I'm going to do another free pass on everything at some point but we can address anything I find in separate small PRs.
